### PR TITLE
View media submissions in submission details

### PR DESCRIPTION
**Pull request by shichongrui**
refs: MBL-11315
affects: Student
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  One or more files in Core are below the minimum test coverage 50%
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please run `yarn update-headers`. The following do not have the correct license header: 
* Core/CoreTests/Database/Models/MediaCommentTests.swift
* TestsFoundation/TestsFoundation/Fixtures/Api/APIMediaCommentFixture.swift
* TestsFoundation/TestsFoundation/Fixtures/Model/MediaCommentFixture.swift
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-11315](https://instructure.atlassian.net/browse/MBL-11315)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 93.18% | 93.36% | -0.19%
Core/Core/Database/CoreData/Models/MediaComment.swift | 0% | -- | --
Core/Core/Extensions/OperationQueueExtensions.swift | 0% | 0% | 0%
**Student** | 94.61% | 94.68% | -0.07%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.54% | 92.54% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Student/Student/Submissions/SubmissionDetails/SubmissionDetailsPresenter.swift [hunk 1]
```diff
@@ -160,6 +161,15 @@ class SubmissionDetailsPresenter {
...
             return UrlSubmissionContentViewController.create(submission: submission)
+        case .some(.media_recording):
+            guard let mediaComment = submission.mediaComment else {
+                return nil
+            }
+            let player = AVPlayer(url: mediaComment.url)
+            let controller = AVPlayerViewController()
+            controller.player = player
+            controller.view.accessibilityIdentifier = "SubmissionDetailsPage.mediaPlayer"
+            return controller
```
Comment (vanwagonet) [inline Student/Student/Submissions/SubmissionDetails/SubmissionDetailsPresenter.swift]: nice. I was worried this was going to be harder.

File: Student/StudentUITests/Submissions/SubmissionDetails/SubmissionDetailsTests.swift [hunk 2]
```diff
@@ -219,4 +219,20 @@ class SubmissionDetailsTests: StudentTest {
...
+            assignment: assignment,
+            context: ContextModel(.course, id: course.id),
+            as: student,
+            submissionType: .media_recording,
+            mediaCommentID: file.media_entry_id,
+            mediaCommentType: .audio
+        )
+        launch("/courses/\(course.id)/assignments/\(assignment.id)", as: student)
+        AssignmentDetailsPage.tap(.viewSubmissionButton)
+        page.assertVisible(.mediaPlayer)
```
Comment (vanwagonet) [inline Student/StudentUITests/Submissions/SubmissionDetails/SubmissionDetailsTests.swift]: Great work on UI testing!

File: TestsFoundation/TestsFoundation/DataSeeding/SeedClient.swift [hunk 3]
```diff
@@ -256,6 +256,19 @@ public struct SeedClient {
...
+        url: URL,
+        named name: String? = nil,
+        as user: AuthUser
+    ) -> APIFile {
+        let target = makeRequest(PostFileUploadTargetRequest(
+            target: .myFiles,
+            body: .init(name: name ?? url.lastPathComponent, on_duplicate: .rename, parent_folder_id: nil)
+        ), with: user.token)
+        return makeRequest(PostFileUploadRequest(fileURL: url, target: target), with: user.token)
+    }
```
Comment (vanwagonet) [inline TestsFoundation/TestsFoundation/DataSeeding/SeedClient.swift]: Does the above `uploadFile` not work for AV? I would assume we want user's assignment submissions to not require using their personal storage quota.

**Pull request merged**